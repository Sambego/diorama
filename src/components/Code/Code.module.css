.code {
  padding: 1rem;
  min-width: 80vw;
  max-width: 80vw;

  background: var(--color-gray-dark);

  box-shadow: 1.5rem 1.5rem 5rem 0 rgba(27, 43, 52, 0.2),
    -1.5rem -1.5rem 5rem 0 rgba(27, 43, 52, 0.2), -1.5rem 1.5rem 5rem 0 rgba(27, 43, 52, 0.2),
    1.5rem -1.5rem 5rem 0 rgba(27, 43, 52, 0.2);
  border-radius: 0.5rem;

  text-align: left;

  code {
    display: block;
  }

  pre {
    padding: 0;
    margin: 0;
    background: none;
    text-align: left;
  }
}

.header {
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  padding: 0 1rem;
  text-align: left;
}

.red,
.yellow,
.green {
  align-self: center;
  display: inline-block;
  width: 1.2rem;
  height: 1.2rem;
  margin: 0 0.5rem 0 0;
  border-radius: 50%;
}

.red {
  background: var(--color-red);
}

.yellow {
  background: var(--color-yellow);
}

.green {
  background: var(--color-green);
}

.lang {
  display: inline-block;
  flex: 1;
  text-align: center;
  color: white;
}

.snippet {
  padding: 2rem 1rem 1rem;
  code,
  pre {
    color: var(--color-gray-6);
    font-size: 2vw;
    line-height: 120%;
  }
}

@media (min-width: 600px) {
  .code {
    min-width: 50vw;
    max-width: 80vw;
  }
}

/* Overwrite the ugly standard prism styles */
span[class~='token'] {
  background: none !important;
}

span[class~='comment'],
span[class~='prolog'],
span[class~='doctype'],
span[class~='cdata'] {
  color: var(--color-gray-3) !important;
}

span[class~='tag'] {
  color: var(--color-gray-4) !important;
}

span[class~='punctuation'] {
  color: var(--color-gray-5) !important;
}

span[class~='operator'],
span[class~='boolean'],
span[class~='number'],
span[class~='attr-name'] {
  color: var(--color-orange) !important;
}

span[class~='property'] {
  color: var(--color-yellow) !important;
}

span[class~='string'],
span[class~='entity'],
span[class~='url'],
span[class~='statement'],
span[class~='regex'],
span[class~='atrule'] {
  color: var(--color-teal) !important;
}

span[class~='selector'] {
  color: var(--color-pink) !important;
}

span[class~='attr']-value,
span[class~='keyword'],
span[class~='control'],
span[class~='directive'],
span[class~='unit'] {
  color: var(--color-green) !important;
}

span[class~='placeholder'],
span[class~='variable'],
span[class~='function'] {
  color: var(--color-blue) !important;
}

span[class~='important'] {
  color: var(--color-red) !important;
}
