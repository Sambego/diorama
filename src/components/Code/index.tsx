import React from "react";
import PropTypes from "prop-types";
import Prism from "prismjs";

import "prismjs/themes/prism.css";
import styles from "./Code.module.css";

type CodeProps = React.PropsWithChildren<{
	className?: string;
	style?: React.CSSProperties;
	code: string;
	lang?: string;
}>;

/* eslint-disable react/no-danger */
function Code({ code, lang, style, className }: CodeProps) {
	return (
		<div className={`${styles.code} diorama-code ${className}`} style={style}>
			<header className={`${styles.header} diorama-code-header`}>
				<span className={styles.red} />
				<span className={styles.yellow} />
				<span className={styles.green} />
				<span className={styles.lang}>{lang}</span>
			</header>
			<div className={`${styles.snippet} diorama-code-snippet`}>
				<pre>
					<code
						dangerouslySetInnerHTML={{
							__html: lang
								? Prism.highlight(
										code,
										Prism.languages[lang],
										// Prism.languages[lang]
										lang
								  )
								: "",
						}}
					/>
				</pre>
			</div>
		</div>
	);
}
/* eslint-enable react/no-danger */

Code.propTypes = {
	className: PropTypes.string,
	code: PropTypes.string.isRequired,
	lang: PropTypes.string,
	style: PropTypes.shape({}),
};

Code.defaultProps = {
	className: "",
	lang: "javascript",
	style: {},
};

export default Code;
