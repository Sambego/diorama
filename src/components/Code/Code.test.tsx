import React from "react";
import renderer from "react-test-renderer";
import Code from ".";

it("renders correctly", () => {
	const tree = renderer
		.create(<Code code="const foo = () => 'bar';" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
