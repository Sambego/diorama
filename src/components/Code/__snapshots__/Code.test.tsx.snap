// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="_code_4ct3i_1 diorama-code "
  style={{}}
>
  <header
    className="_header_4ct3i_27 diorama-code-header"
  >
    <span
      className="_red_4ct3i_35"
    />
    <span
      className="_yellow_4ct3i_36"
    />
    <span
      className="_green_4ct3i_37"
    />
    <span
      className="_lang_4ct3i_58"
    >
      javascript
    </span>
  </header>
  <div
    className="_snippet_4ct3i_65 diorama-code-snippet"
  >
    <pre>
      <code
        dangerouslySetInnerHTML={
          {
            "__html": "<span class=\\"token keyword\\">const</span> <span class=\\"token function-variable function\\">foo</span> <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token operator\\">=></span> <span class=\\"token string\\">'bar'</span><span class=\\"token punctuation\\">;</span>",
          }
        }
      />
    </pre>
  </div>
</div>
`;
